instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcTest.java

```diff
@@ -1669,6 +1677,69 @@ public void sayHello(
     assertThat(error).hasValue(null);
   }
 
+  @Test
+  void setCapturedRequestMetadata() throws Exception {
+    String metadataAttributePrefix = "rpc.request.metadata.";
+    AttributeKey<List<String>> clientAttributeKey =
+        AttributeKey.stringArrayKey(metadataAttributePrefix + CLIENT_REQUEST_METADATA_KEY);
+    AttributeKey<List<String>> serverAttributeKey =
+        AttributeKey.stringArrayKey(metadataAttributePrefix + SERVER_REQUEST_METADATA_KEY);
+    String metadataValue = "some-value";
+    List<String> metadataValueAsList = Collections.singletonList(metadataValue);
+
+    BindableService greeter =
+        new GreeterGrpc.GreeterImplBase() {
+          @Override
+          public void sayHello(
+              Helloworld.Request req, StreamObserver<Helloworld.Response> responseObserver) {
+            Helloworld.Response reply =
+                Helloworld.Response.newBuilder().setMessage("Hello " + req.getName()).build();
+            responseObserver.onNext(reply);
+            responseObserver.onCompleted();
+          }
+        };
+
+    Server server = configureServer(ServerBuilder.forPort(0).addService(greeter)).build().start();
+
+    ManagedChannel channel = createChannel(server);
+
+    Metadata extraMetadata = new Metadata();
+    extraMetadata.put(
+        Metadata.Key.of(SERVER_REQUEST_METADATA_KEY, Metadata.ASCII_STRING_MARSHALLER),
+        metadataValue);
+    extraMetadata.put(
+        Metadata.Key.of(CLIENT_REQUEST_METADATA_KEY, Metadata.ASCII_STRING_MARSHALLER),
+        metadataValue);
```

Comment:
can you use different values for these two metadata? (just for a bit more robustness of the test)

Reply:
@trask Tup, done